synfig-studio/src/gui/workarea.cpp

```diff
@@ -1043,6 +1043,13 @@ WorkArea::on_key_release_event(GdkEventKey* event)
 	if (event_result != Smach::RESULT_OK)
 		return true;
 
+	if (event->keyval == GDK_KEY_Escape && guide_highlighted)
+	{
+		get_guide_list().erase(curr_guide);
+		guide_highlighted = !guide_highlighted;
```

Comment:
based on the `if` condition, just set it to `false`, because the only way this line is reached is `guide_highlighted` being `true`.

Besides, `curr_guide` should now be properly set.

Reply:
couldnt understand the curr_guide part could you explain or give an example. In the case where guide is dragged into the ruler it just erases(curr_guide)

Reply:
The variable scope is not only the method `on_key_release_event()` and after the erase, it stores an invalid iterator, as the container changed size.
Pointing to `get_guide_list().end()` is fine.